### PR TITLE
Document AlertmanagerConfig CRD

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

**Comment by sev3ryn** [comment]
can commit 5a669f3 be merged in another PR. I we noticed that I've done the same in my PR #3692 - it would be great to rebase on that

File: Documentation/api.md [hunk 1]
```diff
@@ -983,166 +982,155 @@ AlertmanagerConfigList is a list of AlertmanagerConfig.
...
+| route | The Alertmanager route definition for alerts matching the resource’s namespace. It will be added to the generated Alertmanager configuration as a first-level route. | *[Route](#route) | false |
+| receivers | List of receivers. | [][Receiver](#receiver) | false |
+| inhibitRules | List of inhibition rules. The rules will only apply to alerts matching the resource’s namespace. | [][InhibitRule](#inhibitrule) | false |
 
 [Back to TOC](#table-of-contents)
 
 ## HTTPConfig
 
-
+HTTPConfig defines a client HTTP configuration.
```
Comment (lilic) [inline Documentation/api.md:997]: Can we add linkio to docs here as well? WDYT? https://prometheus.io/docs/alerting/latest/configuration/#http_config

File: Documentation/api.md [hunk 2]
```diff
@@ -983,166 +982,155 @@ AlertmanagerConfigList is a list of AlertmanagerConfig.
 
 ## AlertmanagerConfigSpec
 
-
+AlertmanagerConfigSpec is a specification of the desired behavior of the Alertmanager configuration. By definition, the Alertmanager configuration only applies to alerts that
```
Comment (lilic) [inline Documentation/api.md:985]: Seems like this was cut mid sentence?

**Review by lilic — COMMENTED** [review]
Nice, thanks so much! Just some suggestions.

File: Documentation/api.md (hunk 1 above)
Comment (simonpasquier) [reply Documentation/api.md]: done

File: Documentation/api.md (hunk 2 above)
Comment (simonpasquier) [reply Documentation/api.md]: nice catch :)

**Comment by simonpasquier** [comment]
@sev3ryn yes I've just seen your PR. If this PR merges first and even without 5a669f35d06fa6bc268561ad6632ee74c23686a6, I suspect that you'll have to deal with merge conflicts anyway.

**Comment by sev3ryn** [comment]
@simonpasquier sure thing, you changed that better, so better to have yours merged earlier

**Review by lilic — APPROVED** [review]
lgtm 🎉

**Pull request merged**